portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,27 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    // public static void main(String[] args){ 
+    // ArrayList<String> favorite = new ArrayList<String>();
+    //     favorite.add("Red");
+    //     favorite.add("Orange");
+    //     favorite.add("Pho");
+        
+    // }
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Rashad!</h1>");
+     ArrayList<String> favorite = new ArrayList<String>();
+        favorite.add("Red");
+        favorite.add("Orange");
+        favorite.add("Pho");
+        String color = favorite.get(0);
+        String fruit = favorite.get(1);
+        String food = favorite.get(2);
+        Gson gson = new Gson();
+        String json = gson.toJson(favorite);
+request.setAttribute("fact", favorite);
```

Comment:
I think the attribute set on request object will get out of scope at the end of request processing in doGet. Were you experimenting with something in this line? What was it?